### PR TITLE
[ch148812] update checkbox

### PR DESCRIPTION
**Pull request by ehoops**
The checkbox updates require using our own SVGs.  This PR adds @svgr/rollup for loading svgs when we build and deploy through npm.  It adds @svgr/webpack for loading svgs in storybook during development.  It would be great to get feedback on this approach.

Checkboxes now have 3 stages: Unchecked, Checked, and Indeterminate.  These are controlled by the parent.  This PR has style updates for checkboxes in each of the stages and in the following states: base, hover, and disabled

Screen Shot 2021-08-13 at 4 06 46 PM
Screen Shot 2021-08-13 at 4 06 55 PM
Screen Shot 2021-08-13 at 4 07 02 PM

Hover states for each:
Screen Shot 2021-08-13 at 4 07 19 PM
Screen Shot 2021-08-13 at 4 07 31 PM
Screen Shot 2021-08-13 at 4 07 25 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,6 +21,8 @@
     "@storybook/addon-essentials": "^6.2.9",
     "@storybook/addon-links": "^6.2.9",
     "@storybook/react": "^6.2.9",
+    "@svgr/rollup": "^5.5.0",
+    "@svgr/webpack": "^5.5.0",
```
Comment (ehoops) [inline package.json:25]: This one is for storybook.

File: src/core/Checkbox/index.tsx [hunk 2]
```diff
@@ -1,10 +1,36 @@
...
+import { ReactComponent as IconCheckboxIndeterminate } from "../../common/svgs/IconCheckboxIndeterminate.svg";
+import { ReactComponent as IconCheckboxUnchecked } from "../../common/svgs/IconCheckboxUnchecked.svg";
+import { getColors } from "../styles";
 
 export { CheckboxProps };
 
-const Checkbox = (props: CheckboxProps): JSX.Element => {
-  return <RawCheckbox {...props} />;
-};
+const StyledCheckbox = styled(RawCheckbox)`
```
Comment (tihuan) [inline src/core/Checkbox/index.tsx:11]: maybe we can move styled components to `./style.ts` for code organization 😄 ?

File: src/core/Checkbox/index.stories.tsx [hunk 3]
```diff
@@ -8,16 +7,92 @@ const actions = {
...
+  <>
+    <p>
+      Accessibility: All checkboxes should have a label. This can be done with a
+      visible form label or with the aria-label attribute.
+      https://material-ui.com/components/checkboxes/#accessibility
+    </p>
+    <Checkbox
+      color="default"
+      defaultChecked
+      inputProps={{ "aria-label": "checkbox-default-checked" }}
```
Comment (tihuan) [inline src/core/Checkbox/index.stories.tsx:20]: nice a11y examples 🔥

**Review by tihuan — APPROVED** [review]
LGTM!! Thanks so much for figuring this out, @ehoops 🙏 Looks awesome!

**Review by j-x-han — APPROVED** [review]
This looks awesome! 🎉 Thanks for adding this
The IconButton changes will also require adding our own svgs, so can't wait to try this out myself 😄

**Comment by tihuan** [comment]
Ah looks like we need to declare a module for `*.svg` files!

example: https://github.com/chanzuckerberg/aspen/blob/trunk/src/frontend/src/common/types/svg.d.ts

**Comment by ehoops** [comment]
Requesting re-review since this now includes the checkbox "stage" changes.  The checkbox is not a controlled component, which seems alright.  

I will document how to add svgs in a follow-up pr, but I think this should not block merging/other PRs

File: src/core/Checkbox/index.stories.tsx [hunk 4]
```diff
@@ -1,23 +1,104 @@
...
+const storyColumn = {
+  alignItems: "flex-start",
+  display: "flex",
+  flexDirection: "column",
+};
+
+storiesOf("Checkbox", module).add("unchecked", () => (
+  <div style={storyColumn as React.CSSProperties}>
+    <p>
+      <strong>Accessibility</strong>: All checkboxes should have a label. This
```
Comment (tihuan) [inline src/core/Checkbox/index.stories.tsx:20]: ooh so informational 💡 ! Great idea!! Thanks for adding this 🤩

File: src/core/Checkbox/index.tsx [hunk 5]
```diff
@@ -1,10 +1,56 @@
-import { Checkbox as RawCheckbox, CheckboxProps } from "@material-ui/core";
+import { CheckboxProps as MUICheckboxProps, SvgIcon } from "@material-ui/core";
 import React from "react";
+import IconCheckboxChecked from "../../common/svgs/IconCheckboxChecked.svg";
+import IconCheckboxIndeterminate from "../../common/svgs/IconCheckboxIndeterminate.svg";
+import IconCheckboxUnchecked from "../../common/svgs/IconCheckboxUnchecked.svg";
+import { StyledCheckbox } from "./styles";
 
-export { CheckboxProps };
+interface CheckboxProps
```
Comment (tihuan) [inline src/core/Checkbox/index.tsx:8]: Let's export the interface too, in case people use it!

```suggestion
export interface CheckboxProps
```

**Review by tihuan — APPROVED** [review]
I like the new `stage` ! Pretty intuitive to use here 🤩 !! Thanks for implementing this, @ehoops 🎉 🚀 🌟 !!

**Comment by Janeece** [comment]
This looks great! 🤩

File: src/core/Checkbox/index.tsx (hunk 5 above)
Comment (ehoops) [reply src/core/Checkbox/index.tsx]: Woah! I've never seen this Github feature

Comment (tihuan) [reply src/core/Checkbox/index.tsx]: Yaas! Magical button 😆 

Screen Shot 2021-08-16 at 11 02 14 AM

**Pull request merged**